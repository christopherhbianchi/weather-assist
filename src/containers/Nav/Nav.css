.nav-bar{
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 50px;
  background: #345fa5;
  box-shadow: 10px 15px 17px #0e0e0e;
}

.nav-link{
  display: block;
  float: left;
  font-family: sans-serif;
  font-size: 11px;
  color: white;
  padding-top: 20px;
  padding-right: 10px;
  padding-bottom: 10px;
  padding-left: 10px;

  text-decoration: none;
}

.nav-link:hover{
  background-color: #17b0cf;
}
