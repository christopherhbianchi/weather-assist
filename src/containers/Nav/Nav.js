import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
//import components below

class Nav extends Component {

  constructor(){
    super();
    this.state = {};
  }

  render(){
    return(
      <nav>
        <NavLink to='/home'>
          Home
        </NavLink>
      </nav>
    );//closes return
  }//closes render

}

export default Nav;
