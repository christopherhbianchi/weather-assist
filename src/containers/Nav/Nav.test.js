import React from 'react';
import { shallow } from 'enzyme';
import { Nav, mapStateToProps, mapDispatchToProps } from './Nav.js';


describe('<Nav />', () => {
  const mockFunction = jest.fn();
  const renderedComponent = shallow(<Nav
    logoutUserSuccess={mockFunction}
    history={
      { push: () => {} }
    }/>);

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });//it


  it('should map the store correctly', () => {
    const mockStore = { activeUser:{} };
    const mapped = mapStateToProps(mockStore);

    expect(mapped).toEqual(mockStore);
  });//it


  it('should dispatch logoutUserSuccess', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.logoutUserSuccess();
    expect(mockDispatch).toHaveBeenCalled();
  });//it


  it('should logout user on click', () => {
    renderedComponent.find('.nav-link-logout').simulate('click', { preventDefault(){} });
    expect(mockFunction.mock.calls.length).toEqual(1);
  });//it

});//describe
