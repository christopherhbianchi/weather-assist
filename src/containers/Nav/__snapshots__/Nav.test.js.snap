// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    history={
        Object {
            "push": [Function],
          }
    }
    logoutUserSuccess={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link"
          to="/home"
>
          WEATHER ASSIST
</NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link-logout"
          onClick={[Function]}
          to="/"
>
          LOG OUT
</NavLink>,
      ],
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "WEATHER ASSIST",
          "className": "nav-link",
          "to": "/home",
        },
        "ref": null,
        "rendered": "WEATHER ASSIST",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "LOG OUT",
          "className": "nav-link-logout",
          "onClick": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": "LOG OUT",
        "type": [Function],
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-link"
            to="/home"
>
            WEATHER ASSIST
</NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-link-logout"
            onClick={[Function]}
            to="/"
>
            LOG OUT
</NavLink>,
        ],
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "WEATHER ASSIST",
            "className": "nav-link",
            "to": "/home",
          },
          "ref": null,
          "rendered": "WEATHER ASSIST",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "LOG OUT",
            "className": "nav-link-logout",
            "onClick": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": "LOG OUT",
          "type": [Function],
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
